tests/solidity/test_fees.py

```diff
@@ -0,0 +1,381 @@
+import unittest
+from typing import Optional, Tuple
+from decimal import Decimal
+
+import pytest
+
+import elfpy.agents.agent as agent
+import elfpy.markets.hyperdrive.hyperdrive_market as hyperdrive_market
+import elfpy.pricing_models.hyperdrive as hyperdrive_pm
+import elfpy.time as time
+import elfpy.types as types
+
+# pylint: disable=duplicate-code
+
+AMOUNT = [10**i for i in range(1, 9)]  # trade amounts up to 100 million
+
+
+class TestFees(unittest.TestCase):
+    """Test case for fees applied to trades"""
+
+    # pylint: disable=too-many-instance-attributes
+
+    contribution: float = 500_000_000
+    target_apr: float = 0.5
+    term_length: int = 365
+    trade_amount: float = 1
+    alice: agent.Agent
+    bob: agent.Agent
+    gary: agent.Agent  # governance gary
+    hyperdrive: hyperdrive_market.Market
+    block_time: time.BlockTime
+    pricing_model: hyperdrive_pm.HyperdrivePricingModel
+
+    def __init__(self, target_apr: Optional[float] = None, gov_fee: float = 0.5, **kwargs):
+        """Set up agent, pricing model, & market for the subsequent tests.
+        This function is run before each test method.
+        """
+        # assign all keyword args to self
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+        if target_apr:
+            self.target_apr = target_apr
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        self.bob.budget = self.trade_amount
+        self.bob.wallet.balance = types.Quantity(amount=self.trade_amount, unit=types.TokenType.BASE)
+        self.gary = agent.Agent(wallet_address=2, budget=0)
+        self.block_time = time.BlockTime()
+        self.pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState(
+            trade_fee_percent=0.1,  # 0.1e18, // curveFee
+            redemption_fee_percent=0.1,  # 0.1e18, //flatFee
+            governance_fee_percent=gov_fee,  # 0.5e18, //govFee
+        )
+        super().__init__()
+
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=self.pricing_model,
+            market_state=market_state,
+            block_time=self.block_time,
+            position_duration=time.StretchedTime(
+                days=self.term_length,
+                time_stretch=self.pricing_model.calc_time_stretch(self.target_apr),
+                normalizing_constant=self.term_length,
+            ),
+        )
+        _, wallet_deltas = self.hyperdrive.initialize(
+            wallet_address=self.alice.wallet.address,
+            contribution=self.contribution,
+            target_apr=self.target_apr,
+        )
+        self.alice.wallet.update(wallet_deltas)
+        self.market_state_before_open = self.hyperdrive.market_state.copy()
+
+    def assertAlmostEqual(self, first, second, places=None, msg=None, delta=None):
```

Comment:
You should not have to cast anything to Decimal. The outputs of the calc functions (trade breakdown), as well as the market state should already be casting everything to a float.

Reply:
everything passes without it lol
i swear this wasn't the case before
removed in 0968416ac0c76e40b2434dee977500c55265e22c